src/badger.ts

```diff
@@ -71,8 +74,42 @@ export abstract class Badger {
    */
   public abstract determineRating(pullRequests: any[], username?: string): number;
 
-  protected ignoreThisUsername(username: string) {
-    return this.ignoreUsernames?.includes(username);
+  protected async ignoreThisUsername(username: string): Promise<boolean> {
+    // Check individual usernames first (backward compatibility)
```

Comment:
i don't think backwards compatibility is really at play here. its just an implementation choice to check provided usernames first. if you checked teams first, you wouldn't be actually changing the behavior of this function at all